console/console-init/ui/src/modules/device/dialogs/CreateDevice/CreateDevice.tsx

```diff
@@ -1,21 +1,174 @@
-import { Wizard } from "@patternfly/react-core";
-import React from "react";
+/*
+ * Copyright 2020, EnMasse authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+
+import React, { useState } from "react";
+import {
+  Wizard,
+  WizardFooter,
+  WizardContextConsumer,
+  Button
+} from "@patternfly/react-core";
 import { DeviceInformation } from "modules/device/components/DeviceInformation";
+import { ConnectionType } from "modules/device/components/ConnectionTypeStep";
 
-export const CreateDevice: React.FunctionComponent<{}> = () => {
+const CreateDevice: React.FunctionComponent<any> = () => {
+  const [isWizardOpen, setIsWizardOpen] = useState<boolean>(true);
+  const [connectionType, setConnectionType] = useState<string>();
+  const onToggle = () => {
+    setIsWizardOpen(!isWizardOpen);
+    resetForm();
+  };
+  const resetForm = () => {};
+  const addGateway = {
+    name: "Add gateways",
+    component: <p>Add gateway</p>
+  };
+  const addCredentials = {
+    name: "Add credentials",
+    component: <p>Add credentials</p>
+  };
+  const reviewForm = {
+    name: "Review",
+    component: <p>Review</p>
+  };
+  const onConnectionChange = (_: boolean, event: any) => {
+    const data = event.target.value;
+    if (data) {
+      setConnectionType(data);
+    }
+  };
   const steps = [
-    { name: "Step 1", component: <DeviceInformation /> },
-    { name: "Step 2", component: <p>Step 2</p> },
-    { name: "Step 3", component: <p>Step 3</p> },
-    { name: "Step 4", component: <p>Step 4</p> },
+    { name: "Device information", component: <DeviceInformation /> },
     {
-      name: "Final Step",
-      component: <p>Final Step</p>,
-      hideCancelButton: true,
-      nextButtonText: "Close"
+      name: "Connection Type",
+      component: (
+        <ConnectionType
+          connectionType={connectionType}
+          onConnectionChange={onConnectionChange}
+        />
+      )
     }
   ];
+  if (connectionType) {
+    if (connectionType === "directly") {
+      steps.push(addCredentials);
+    } else {
+      steps.push(addGateway);
+    }
+    steps.push(reviewForm);
+  }
+
+  const handleNextIsEnabled = () => {
+    return false;
+  };
+
+  const CustomFooter = (
```

Comment:
Can you try and modularize the footer logic here?

Reply:
Yes, I will do that, for now, this is just a dummy component. As there are alot of validation required for footer action, My suggestion would be for now, keep it simple. In future I will modify this component.

Reply:
Okay